File: src/main/java/uk/co/threebugs/mochiwhattotrade3/setupgroups/SetupGroupsController.java
package uk.co.threebugs.mochiwhattotrade3.setupgroups;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class SetupGroupsController {

    private final SetupGroupsService setupGroupsService;

    @GetMapping("/setupGroups")
    public List<SetupGroupsDto> findAll() {
        return setupGroupsService.findAll();
    }

}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/setupgroups/SetupGroupsController.kt
package uk.co.threebugs.mochiwhattotrade3.setupgroups

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class SetupGroupsController (
    private val setupGroupsService: SetupGroupsService
)
{
    @GetMapping("/setupGroups")
    fun findAll(): List<SetupGroupsDto> {
        return setupGroupsService.findAll()
    }
}
